pkg/apis/componentconfig/v1alpha1/temp_v1a1_refactor.gogo

```diff
@@ -0,0 +1,548 @@
+  OOMScoreAdj *int32 `json:"oomScoreAdj"`
+  // A set of ResourceName=Percentage (e.g. memory=50%) pairs that describe
+  // how pod resource requests are reserved at the QoS level.
+  // Currently only memory is supported. [default=none]"
+  ExperimentalQOSReserved map[string]string `json:"experimentalQOSReserved"`
+  // maxPods is the number of pods that can run on this Kubelet.
+  MaxPods int32 `json:"maxPods"`
+  // Maximum number of pods per core. Cannot exceed MaxPods
+  PodsPerCore int32 `json:"podsPerCore"`
+  // cpuCFSQuota is Enable CPU CFS quota enforcement for containers that
+  // specify CPU limits
+  CPUCFSQuota *bool `json:"cpuCFSQuota"`
+}
+
+type CgroupsConfig struct {
+  // kubeletCgroups is the absolute name of cgroups to isolate the kubelet in.
+  KubeletCgroups string `json:"kubeletCgroups"`
+  // systemCgroups is absolute name of cgroups in which to place
+  // all non-kernel processes that are not already in a container. Empty
+  // for no container. Rolling back the flag requires a reboot.
+  SystemCgroups string `json:"systemCgroups"`
+  // runtimeCgroups are cgroups that container runtime is expected to be isolated in.
+  RuntimeCgroups string `json:"runtimeCgroups"` // TODO(kc-refactor): Does this become part of CRI now?
+  // cgroupRoot is the root cgroup to use for pods. This is handled by the
+  // container runtime on a best effort basis.
+  CgroupRoot string `json:"cgroupRoot"`
+  // Enable QoS based Cgroup hierarchy: top level cgroups for QoS Classes
+  // And all Burstable and BestEffort pods are brought up under their
+  // specific top level QoS cgroup.
+  // +optional
+  CgroupsPerQOS *bool `json:"cgroupsPerQOS,omitempty"`
+  // driver that the kubelet uses to manipulate cgroups on the host (cgroupfs or systemd)
+  // +optional
+  CgroupDriver string `json:"cgroupDriver,omitempty"`
+}
+
+type NodeAllocatableConfig struct {
+  // A set of ResourceName=ResourceQuantity (e.g. cpu=200m,memory=150G) pairs
+  // that describe resources reserved for non-kubernetes components.
+  // Currently only cpu and memory are supported. [default=none]
+  // See http://kubernetes.io/docs/user-guide/compute-resources for more detail.
+  SystemReserved map[string]string `json:"systemReserved"`
+  // A set of ResourceName=ResourceQuantity (e.g. cpu=200m,memory=150G) pairs
+  // that describe resources reserved for kubernetes system components.
+  // Currently only cpu and memory are supported. [default=none]
+  // See http://kubernetes.io/docs/user-guide/compute-resources for more detail.
+  KubeReserved map[string]string `json:"kubeReserved"`
+  // This flag helps kubelet identify absolute name of top level cgroup used to enforce `SystemReserved` compute resource reservation for OS system daemons.
+  // Refer to [Node Allocatable](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node-allocatable.md) doc for more information.
+  SystemReservedCgroup string `json:"systemReservedCgroup,omitempty"`
```

Comment:
this is where the separate struct for CgroupsConfig feels awkward.  can we combine them into a common NodeAllocatableCnofig struct?